lib/manageiq/password.rb

```diff
@@ -23,58 +22,46 @@ def initialize(str = nil)
       @encStr = encrypt(str)
     end
 
-    def encrypt(str, ver = "v2", key = self.class.keys[ver])
-      value = key.encrypt64(str).delete("\n") unless str.nil? || str.empty?
-      "#{ver}:{#{value}}"
-    end
+    def encrypt(str, key = self.class.key)
+      return str if str.nil?
 
-    def decrypt(str, legacy = false)
-      if str.nil? || str.empty?
-        str
-      else
-        ver, enc = self.class.split(str)
-        return "" if enc.empty?
+      enc = key.encrypt64(str).delete("\n") unless str.empty?
+      wrap(enc)
+    end
 
-        ver ||= "0" # if we don't know what it is, just assume legacy
-        key_name = (ver == "2" && legacy) ? "alt" : "v#{ver}"
+    def decrypt(str, key = self.class.key)
+      enc = unwrap(str)
+      return enc if enc.nil? || enc.empty?
 
-        begin
-          self.class.keys[key_name].decrypt64(enc).force_encoding('UTF-8')
-        rescue
-          raise PasswordError, "can not decrypt v#{ver}_key encrypted string"
-        end
+      begin
+        key.decrypt64(enc).force_encoding('UTF-8')
+      rescue
+        raise PasswordError, "cannot decrypt encrypted string"
       end
     end
 
     def recrypt(str)
-      return str if str.nil? || str.empty?
-      decrypted_str =
-        begin
-          # if a legacy v2 key exists, give decrypt the option to use that
-          decrypt(str, self.class.keys["alt"])
-        rescue
-          source_version = self.class.split(str).first || "0"
-          if source_version == "0" # it probably wasn't encrypted
-            return str
-          elsif source_version == "2" # tried with an alt key, see if regular v2 key works
-            decrypt(str)
-          else
-            raise
-          end
-        end
+      return str if str.nil?
+
+      decrypted_str   = decrypt(str, self.class.legacy_keys["alt"]) rescue nil
```

Comment:
thanks for making this change.

I was expecting you to keep the options as the input. Have it override alt if present.
Then you could push through this, fix all the callers to use the better interface, then drop the legacy/alt interface.
0 lock step changes.

this works too

Reply:
I'm not likely going to lockstep this in...seems like a lot of overhead when the goal is the newer interface i have in mind.